src/chat.rs

```diff
@@ -1041,7 +1041,9 @@ impl Chat {
         if chat.id.is_archived_link() {
             chat.name = stock_str::archived_chats(context).await;
         } else {
-            if chat.typ == Chattype::Single {
+            if chat.typ == Chattype::Single && chat.name.is_empty() {
+                // chat.name is set to contact.display_name on changes,
+                // however, if things went wrong somehow, we do this here explicitly.
```

Comment:
unsure here - we could also just strike this block, however, in this case, we may have empty chatnames if things went wrong at some point in the past. as this is highly visible, i think, we should leave that as a fallback.

for non-empty but wrong names (also from some bug in the past): it is better to show the wrong name in this case, it can be updated and is also searchable. otherwise, one may end up with a chatname that is not searchable again (if we would always return get_displayname)